File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/legacy/rest/model/publicroom/PublicRoom.java
/*
 * Copyright 2014 OpenMarket Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package im.vector.matrix.android.internal.legacy.rest.model.publicroom;

import com.google.gson.annotations.SerializedName;

import java.util.List;

/**
 * Class representing the objects returned by /publicRooms call.
 */
public class PublicRoom {

    public List<String> aliases;

    @SerializedName("canonical_alias")
    public String canonicalAlias;

    public String name;

    // number of members which have joined the room (the members list is not provided)
    @SerializedName("num_joined_members")
    public int numJoinedMembers;

    @SerializedName("room_id")
    public String roomId;

    public String topic;

    // true when the room history is visible (room preview)
    @SerializedName("world_readable")
    public boolean worldReadable;

    // a guest can join the room
    @SerializedName("guest_can_join")
    public boolean guestCanJoin;

    @SerializedName("avatar_url")
    public String avatarUrl;
}

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/api/session/room/model/roomdirectory/PublicRoom.kt
/*
 * Copyright 2014 OpenMarket Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package im.vector.matrix.android.api.session.room.model.roomdirectory

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

/**
 * Class representing the objects returned by /publicRooms call.
 */
@JsonClass(generateAdapter = true)
data class PublicRoom(

        /**
         * Aliases of the room. May be empty.
         */
        @Json(name = "aliases")
        var aliases: List<String>? = null,

        /**
         * The canonical alias of the room, if any.
         */
        @Json(name = "canonical_alias")
        var canonicalAlias: String? = null,

        /**
         * The name of the room, if any.
         */
        @Json(name = "name")
        var name: String? = null,

        /**
         * Required. The number of members joined to the room.
         */
        @Json(name = "num_joined_members")
        var numJoinedMembers: Int = 0,

        /**
         * Required. The ID of the room.
         */
        @Json(name = "room_id")
        var roomId: String,

        /**
         * The topic of the room, if any.
         */
        @Json(name = "topic")
        var topic: String? = null,

        /**
         * Required. Whether the room may be viewed by guest users without joining.
         */
        @Json(name = "world_readable")
        var worldReadable: Boolean = false,

        /**
         * Required. Whether guest users may join the room and participate in it. If they can,
         * they will be subject to ordinary power level rules like any other user.
         */
        @Json(name = "guest_can_join")
        var guestCanJoin: Boolean = false,

        /**
         * The URL for the room's avatar, if one is set.
         */
        @Json(name = "avatar_url")
        var avatarUrl: String? = null,

        /**
         * Undocumented item
         */
        @Json(name = "m.federate")
        var isFederated: Boolean = false

)
